runhouse/resources/hardware/cluster_factory.py

```diff
@@ -262,14 +262,17 @@ def kubernetes_cluster(
         except subprocess.CalledProcessError as e:
             logger.info(f"Error setting context: {e}")
 
-    c = OnDemandCluster(
-        name=name,
-        instance_type=instance_type,
-        provider="kubernetes",
-        launcher_type=launcher_type,
-        server_connection_type=server_connection_type,
-        **kwargs,
-    )
+    try:
+        c = Cluster.from_name(name=name)
```

Comment:
decided to have `Cluster.from_name(name=name)` instead of `OnDemandCluster.from_name` for constancy in the cluster_factory.py.